cluster/gce/gci/configure-helper.sh

```diff
@@ -66,8 +68,57 @@ function create-dirs {
   fi
 }
 
-# Formats the given device ($1) if needed and mounts it at given mount point
-# ($2).
+function get-local-disk-num() {
+  interface=$1
+  format=$2
+
+  total=0
+  if [[ ! -z ${NODE_LOCAL_SSDS_EXT} ]]; then
+    IFS=";" read -r -a ssdgroups <<< ${NODE_LOCAL_SSDS_EXT}
+    for ssdgroup in "${ssdgroups[@]}"
+    do
+      IFS="," read -r -a ssdopts <<< ${ssdgroup}
+      opnum=${ssdopts[0]}
+      opinterface=${ssdopts[1]}
+      opformat=${ssdopts[2]}
+
+      if [[ ${opformat,,} == ${format,,} && ${opinterface,,} == ${interface,,} ]]; then
+        total=$((total+opnum))
+      fi
+    done
+  fi
+  
+}
+
+function safe-block-symlink(){
+  device=$1
+  interface=$2
+  mkdir -p "${UUID_MNT_PREFIX}-${interface,,}-block"
+  ssdmap="/home/kubernetes/localssdmap.txt"
+  echo "Creating symlink for block devices..."
+
+  if [[ ! -e ${ssdmap} ]]; then
+    touch ${ssdmap}
+    chmod +w ${ssdmap}
+  fi
+
+  # each line looks like "${device} persistent-uuid"
+  if [[ ! -z $(cat ${ssdmap} | grep ${device}) ]]; then
+    #create symlink based on saved uuid
+    myuuid=$(cat ${ssdmap} | grep ${device} | cut -d ' ' -f 2)
+  else
+    # generate new uuid and add it to the map
+    myuuid=$(uuidgen)
+    echo "${device} ${myuuid}" >> ${ssdmap}
+  fi
+
+  sym="${UUID_MNT_PREFIX}-${interface,,}-block/local-ssd-${myuuid}"
+  # dont make the exact directory first or else symlink goes 1 directory down
```

Comment:
can you explain this comment?  i'm not understanding it

Reply:
If you make the directory with "mkdir" before doing a symlink to that directory, it actually symlinks to "inside" the directory instead of "to" the directory. I've updated the comment to:

# Do not "mkdir -p ${sym}" as that will cause unintended symlink behavior

Reply:
Ok yeah that's correct.  Symlinks are files, not directories.

Reply:
that explains a lot